Missing overload Assert.That(TestDelegate, IResolveConstraint, string, params object[])

The Assert.That(TestDelegate, ...) overload allows to pass in a delegate as follows:
`Assert.That(() => { Console.Write(null, null); }, Throws.TypeOf<ArgumentNullException>());`
However writing the assertion statement with a message
`Assert.That(() => { Console.Write(null, null); }, Throws.TypeOf<ArgumentNullException>(), "A message");`
yields the compiler errors:
(NUnit 3.0.0 beta 2)
error CS1643: Not all code paths return a value in lambda expression of type 'System.Func<bool>'
error CS1660: Cannot convert lambda expression to type 'bool' because it is not a delegate type
(NUnit 2.6.4)
error CS1660: Cannot convert lambda expression to type 'object' because it is not a delegate type.

This is due to the missing Assert.That(TestDelegate, IResolveConstraint, string, params object[]) overload.

The workaround is to wrap the anonymous method into a TestDelegate:
`Assert.That(new TestDelegate(() => { Console.Write(null, null); }), Throws.TypeOf<ArgumentNullException>(), "A message");`

The cast adds unnecessary "noise" to the statement and the compiler errors are unexpected and confusing when encountered the first time.

--- a/src/NUnitFramework/framework/Assert.That.cs
+++ b/src/NUnitFramework/framework/Assert.That.cs
@@ -130,7 +130,20 @@ namespace NUnit.Framework
         /// <param name="constraint">A ThrowsConstraint used in the test</param>
         static public void That(TestDelegate code, IResolveConstraint constraint)
         {
-            Assert.That((object)code, constraint);
+            Assert.That(code, constraint, null, null);
+        }
+
+        /// <summary>
+        /// Asserts that the code represented by a delegate throws an exception
+        /// that satisfies the constraint provided.
+        /// </summary>
+        /// <param name="code">A TestDelegate to be executed</param>
+        /// <param name="constraint">A ThrowsConstraint used in the test</param>
+        /// <param name="message">The message that will be displayed on failure</param>
+        /// <param name="args">Arguments to be used in formatting the message</param>
+        static public void That(TestDelegate code, IResolveConstraint constraint, string message, params string[] args)
+        {
+            Assert.That((object)code, constraint, message, args);
         }
 
         #endregion

--- a/src/NUnitFramework/tests/Assertions/AssertFailTests.cs
+++ b/src/NUnitFramework/tests/Assertions/AssertFailTests.cs
@@ -39,6 +39,15 @@ namespace NUnit.Framework.Assertions
         }
 
         [Test]
+        public void ThrowsAssertionException_MessageSpecified()
+        {
+            Assert.That(
+                () => Assert.Fail(),
+                Throws.TypeOf<AssertionException>(),
+                "My Message");
+        }
+
+        [Test]
         public void ThrowsAssertionExceptionWithMessage()
         {
             Assert.That(

